app/Services/MetaData.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use DOMDocument;
+use Exception;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Str;
+
+final readonly class MetaData
+{
+    /**
+     * The Open Graph data.
+     *
+     * @var Collection<string, string>
+     */
+    private Collection $data;
+
+    /**
+     * Fetch the Open Graph data for a given URL.
+     */
+    public function __construct(private string $url)
+    {
+        /** @var Collection<string, string> $cachedData */
+        $cachedData = Cache::remember(
```

Comment:
can you lazy evaluate this? so basically the code looks like this:
```

$metaData = new MetaData('my-url');
$metaData->fetch(); // calls are done here...

Reply:
Done, however needed to either store the value of the fetch calls in a seperate variable, so we can then check if there collection is empty, etc, or store have fetch update the data property, then have a seperate getter for it.

Wasn't sure which way you'd prefer, so I went with the simplest, however can change easily.

```php
$service = new MetaData($url);
$metadata = $service->fetch();
```